include/xtensor/xbuffer_adaptor.hpp

```diff
@@ -658,6 +671,21 @@ namespace xt
             swap(p_data, rhs.p_data);
             swap(m_size, rhs.m_size);
         }
+
+        template <class CP, class A>
+        template <class P>
+        inline void xbuffer_storage<CP, A>::replace(P&& data) noexcept
```

Comment:
This method should be removed, the API should force the user to pass the size to avoid inconsistencies.